### PR TITLE
react-bootstrap v0.30.0

### PR DESCRIPTION
**Pull request by bat-bold**
case 1. Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

### CODE REVIEW

File: react-bootstrap/react-bootstrap.d.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 // Type definitions for react-bootstrap
 // Project: https://github.com/react-bootstrap/react-bootstrap
-// Definitions by: Walker Burgin <https://github.com/walkerburgin>, Vincent Siao <https://github.com/vsiao>, Danilo Barros <https://github.com/danilojrr>
+// Definitions by: Walker Burgin <https://github.com/walkerburgin>,
+// Vincent Siao <https://github.com/vsiao>, Danilo Barros <https://github.com/danilojrr>,
```
Comment (mhegazy) [inline react-bootstrap/react-bootstrap.d.ts]: this needs to be on the same line i believe for the dt-bot to know how to parse it.

File: react-bootstrap/react-bootstrap.d.ts [hunk 2]
```diff
@@ -72,7 +74,7 @@ declare namespace ReactBootstrap {
     {
         active?: boolean;
         id?: string | number;
-        linkId?: string | number;
+        // linkId?: string | number; // TODO: Removed since v0.30.0
```
Comment (mhegazy) [inline react-bootstrap/react-bootstrap.d.ts]: `@deprecated` instead

**Comment by mhegazy** [comment]
@walkerburgin @vsiao @danilojrr any comments?

File: react-bootstrap/react-bootstrap.d.ts [hunk 3]
```diff
@@ -444,8 +447,8 @@ declare namespace ReactBootstrap {
         bsSize?: Sizes;
         bsStyle?: string;
         placement?: string;
-        positionLeft?: number;
-        positionTop?: number;
+        positionLeft?: number|string; // String support added since v0.30.0
+        positionTop?: number|string; // String support added since v0.30.0
```
Comment (vsiao) [inline react-bootstrap/react-bootstrap.d.ts]: nit: spaces around `|` please

File: react-bootstrap/react-bootstrap.d.ts [hunk 4]
```diff
@@ -588,6 +591,7 @@ declare namespace ReactBootstrap {
     interface NavDropdownBaseProps
         extends DropdownBaseProps
     {
+        active?:boolean;
```
Comment (vsiao) [inline react-bootstrap/react-bootstrap.d.ts]: missing space after `:`

File: react-bootstrap/react-bootstrap.d.ts [hunk 5]
```diff
@@ -652,13 +656,15 @@ declare namespace ReactBootstrap {
 
 
     // <Pager />
-    interface PagerProps
-        extends React.HTMLProps<Pager>
+    interface PagerProps extends React.HTMLProps<Pager>
```
Comment (vsiao) [inline react-bootstrap/react-bootstrap.d.ts]: could you also put `{` on the same line here?

File: react-bootstrap/react-bootstrap.d.ts [hunk 6]
```diff
@@ -890,44 +898,45 @@ declare namespace ReactBootstrap {
...
+    //     hasFeedback?: boolean;
+    //     help?: any; // TODO: Add more specific type
+    //     labelClassName?: string;
+    //     wrapperClassName?: string;
+    // }
     /** @deprecated since v0.29.0 */
-    class ButtonInput extends React.Component<ButtonInputProps, {}> {
-    }
+    // class ButtonInput extends React.Component<ButtonInputProps, {}> {
+    // }
```
Comment (vsiao) [inline react-bootstrap/react-bootstrap.d.ts]: You should just delete this—we have version control if we ever want it back.

File: react-bootstrap/react-bootstrap-tests.jsx [hunk 7]
```diff
@@ -0,0 +1,959 @@
+// React-Bootstrap Test
```
Comment (vsiao) [inline react-bootstrap/react-bootstrap-tests.jsx]: We probably don't want to check in `.jsx` files in addition to `.tsx`, right?

File: react-bootstrap/react-bootstrap-tests.tsx [hunk 8]
```diff
@@ -7,7 +7,20 @@
 // --------------------------------------------------------------------------------
 import * as React from 'react';
 import { Component, CSSProperties } from 'react';
-import { Button, ButtonToolbar, Modal, Well, ButtonGroup, DropdownButton, MenuItem, Panel, ListGroup, ListGroupItem, Accordion, Tooltip, OverlayTrigger, Popover, ProgressBar, Nav, NavItem, Navbar, NavDropdown, Tabs, Tab, Pager, PageItem, Pagination, Alert, Carousel, CarouselItem, Grid, Row, Col, Thumbnail, Image, Label, Badge, Jumbotron, PageHeader, Glyphicon, Table, Input, ButtonInput, Form, FormGroup, ControlLabel, FormControl, HelpBlock, Radio, Checkbox, Media } from 'react-bootstrap';
+import {
+  Button, ButtonToolbar, Modal, Well, ButtonGroup,
```
Comment (yuit) [inline react-bootstrap/react-bootstrap-tests.tsx]: Seems like it should be 4 space here

**Pull request merged**